File: HackerRank/TimeFormat.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'timeConversion' function below.
     *
     * The function is expected to return a STRING.
     * The function accepts STRING s as parameter.
     */

    public static String timeConversion(String s)
            throws ParseException

    {
        // Write your code here

        DateFormat dateformat =  new SimpleDateFormat("hh:mm:ssaa");
        DateFormat formate = new SimpleDateFormat("HH:mm:ss");
        Date time = null;
        String output = "";

        time = dateformat.parse(s);

        output = formate.format(time);
        return output;

    }
}
public class Solution
{
    public static void main(String[] args) throws IOException,ParseException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        String s = bufferedReader.readLine();

        String result = Result.timeConversion(s);

        bufferedWriter.write(result);
        bufferedWriter.newLine();

        bufferedReader.close();
        bufferedWriter.close();
    }
}


Input (stdin)

07:05:45PM
Your Output (stdout)

19:05:45
Expected Output

19:05:45

File: Problem Solving/TIme-format.py
import math
import os
import random
import re
import sys


def timeConversion(s):
    h=int(s[0:2])
    
    if s[-2]=='P' or s[-2]=='p':
        if h!=12:
            h=12+h
        else:
            h=h    
        return(str(h)+s[2:8])
    else:
        if h==12:
            return("00"+s[2:8])
        else:
            return(s[:8])

    
    # Write your code here

if __name__ == '__main__':
    
    s = input()

    result = timeConversion(s)

    print(result + '\n')


